src/tests/reconnect.test.ts

```diff
@@ -0,0 +1,76 @@
+import FakeTimers from '@sinonjs/fake-timers'
+import test from 'ava'
+import Docker from 'dockerode'
+import { createClient } from 'redis'
+import transporter from '../index.js'
+
+const emit = () => undefined
+
+test.serial(
+  'should be able to reconnect to Redis if the server has disconnected',
+  async (t) => {
+    const clock = FakeTimers.install({ toFake: ['Date'], now: Date.now() })
```

Comment:
You can drop the whole FakeTimers usage if you want to. I haven't been able to find a way to progress the timer so that the `network.connect()` that restores the Redis connection is called slightly after the `redisClient.quit()` function is called. So all though it might be flaky, I don't think you have any other choice but using "real" timeouts and sleeps in this test.